dashboards-observability/public/components/explorer/log_explorer.tsx

```diff
@@ -78,8 +79,9 @@ export const LogExplorer = ({
     }
 
     const index: number = tabIds.indexOf(TabIdToBeClosed);
+    const curSelectedTab = curSelectedTabIdRef.current;
     let newIdToFocus = '';
-    if (TabIdToBeClosed === curSelectedTabId) {
+    if (TabIdToBeClosed === curSelectedTab) {
```

Comment:
question, could you explain a bit why curSelectedTabId and its ref are different here?

Reply:
The handTabClose was referring to the stale tabId when it was attached to the UI. When adding the new tab, the current selected tabId would always set to id of the newly added tab, and when you click on any tab before it, the clicked tab data is passed to the handleTabClick handler, and setting the redux store correctly, but since previously the handTabClose is still referring to the stale curSelectedTabId instead of the updated select tabId, it causes this error.

Reply:
Thanks, got it. I forgot that the tabs are memorized, so this should also be fixed if `curSelectedTabId` is added as a dependency of `memorizedTabs`